pkg/controller/dataimportcron-format-handlers.go

```diff
@@ -0,0 +1,141 @@
+	dataSourceName := dataImportCron.Spec.ManagedDataSource
+	digest := dataImportCron.Annotations[AnnSourceDesiredDigest]
+	if digest == "" {
+		return nil
+	}
+	dvName, err := createDvName(dataSourceName, digest)
+	if err != nil {
+		return err
+	}
+
+	className, err := cc.GetSnapshotClassForSmartClone(dataVolume.Name, &dvStorageClass.Name, r.log, r.client)
+	if err != nil {
+		return err
+	}
+	labels := map[string]string{
+		common.CDILabelKey:       common.CDILabelValue,
+		common.CDIComponentLabel: "",
+	}
+	desiredSnapshot := &snapshotv1.VolumeSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dvName,
+			Namespace: dataImportCron.Namespace,
+			Labels:    labels,
+		},
+		Spec: snapshotv1.VolumeSnapshotSpec{
+			Source: snapshotv1.VolumeSnapshotSource{
+				PersistentVolumeClaimName: &dvName,
+			},
+			VolumeSnapshotClassName: &className,
+		},
+	}
+	r.setDataImportCronResourceLabels(dataImportCron, desiredSnapshot)
+
+	currentSnapshot := &snapshotv1.VolumeSnapshot{}
+	if err := r.client.Get(context.TODO(), client.ObjectKeyFromObject(desiredSnapshot), currentSnapshot); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		cc.AddAnnotation(desiredSnapshot, AnnLastUseTime, time.Now().UTC().Format(time.RFC3339Nano))
+		if err := r.client.Create(ctx, desiredSnapshot); err != nil {
+			return err
+		}
+		return nil
+	}
+	if cc.IsSnapshotReady(currentSnapshot) {
+		// Clean up DV/PVC as they are not needed anymore
+		r.log.Info("Deleting dv/pvc as snapshot is ready", "name", desiredSnapshot.Name)
+		if err := r.deleteDvPvc(ctx, desiredSnapshot.Name, desiredSnapshot.Namespace); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (snapshotFormatHandler) updateOnSuccess(ctx context.Context, r *DataImportCronReconciler, dataImportCron *cdiv1.DataImportCron, sourceObj client.Object) {
+	snapshot := sourceObj.(*snapshotv1.VolumeSnapshot)
+	if snapshot == nil {
+		// Snapshot create/update will trigger reconcile
+		return
+	}
+	if cc.IsSnapshotReady(snapshot) {
+		updateDataImportCronCondition(dataImportCron, cdiv1.DataImportCronUpToDate, corev1.ConditionTrue, "Latest import is up to date", upToDate)
+	} else {
+		updateDataImportCronCondition(dataImportCron, cdiv1.DataImportCronUpToDate, corev1.ConditionFalse, "Snapshot of imported data is progressing", inProgress)
+	}
+}
+
+type pvcFormatHandler struct{}
+
+func (pvcFormatHandler) populateDataSourceSpec(dataSourceSpec *cdiv1.DataSourceSpec, sourcePVC *cdiv1.DataVolumeSourcePVC) {
+	dataSourceSpec.Source = cdiv1.DataSourceSource{
+		PVC: sourcePVC,
+	}
+}
+
+func (pvcFormatHandler) handleSourceCreation(context.Context, *DataImportCronReconciler, *cdiv1.DataImportCron, *cdiv1.DataVolume, *storagev1.StorageClass) error {
+	// No extra steps needed
+	return nil
+}
+
+func (pvcFormatHandler) updateOnSuccess(_ context.Context, _ *DataImportCronReconciler, dataImportCron *cdiv1.DataImportCron, _ client.Object) {
```

Comment:
Same here, I think the interface should not be passing the context or the reconciler, and then the empty parameters are not needed.

Reply:
yep, thx